### PR TITLE
AIP-203: Field behavior / optional format

### LINKED ISSUE #178: AIP-203: Field behavior / optional format
Old lint rule: https://g3doc.corp.google.com/google/api/tools/linter/g3doc/rules/optional-format.md?cl=head

### PR DESCRIPTION
**Pull request by michaelwq**
1）AIP-203: Field behavior / optional format
2）Fix the rule uri and small typo in aip203

Fixes #178.

### CODE REVIEW

File: rules/aip0203/optional.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+			return []lint.Problem{{
+				Message:    "Field behavior `(google.api.field_behavior) = OPTIONAL` shouldn't be used together with other field behaviors.",
+				Descriptor: f,
+			}}
+		}
+		return nil
+	},
+}
+
+// If a field is described as optional, ensure that every optional field on the
```
Comment (mingzhi) [inline rules/aip0203/optional.go]: Is it a typo in `If a field`? Should it start with `If a message`?

File: rules/aip0203/optional.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+func withoutOptionalFieldBehavior(f *desc.FieldDescriptor) bool {
+	for _, v := range utils.GetFieldBehavior(f) {
+		if v == annotations.FieldBehavior_OPTIONAL {
+			return false
+		}
+	}
+	return true
+}
+
+func hasOptionalFieldBehavior(m *desc.MessageDescriptor) bool {
```
Comment (mingzhi) [inline rules/aip0203/optional.go]: Suggest to use `messageHasOptionalFieldBehavior`.

**Comment by lukesneeringer** [comment]
Also, @michaelwq, after the next merge, can you pull the master branch before starting the next thing? You have an ever-increasing number of irrelevant commits.

File: rules/aip0203/optional.go (hunk 2 above)
Comment (michaelwq) [reply rules/aip0203/optional.go]: Good idea. Done.

File: rules/aip0203/optional.go (hunk 1 above)
Comment (michaelwq) [reply rules/aip0203/optional.go]: To make it more clear, change it "If a message has a field which is described as optional"

**Comment by michaelwq** [comment]
> Also, @michaelwq, after the next merge, can you pull the master branch before starting the next thing? You have an ever-increasing number of irrelevant commits.

**Comment by michaelwq** [comment]
> Also, @michaelwq, after the next merge, can you pull the master branch before starting the next thing? You have an ever-increasing number of irrelevant commits.

Sure. thanks Luke.

**Pull request merged**